client/state/themes/utils.js

```diff
@@ -190,6 +191,22 @@ export function getSerializedThemesQueryWithoutPage( query, siteId ) {
 }
 
 /**
+ * Filter wpcom themes from Jetpack theme list
+ * Themes on Jetpack installed from WordPress.com have -wpcom suffix.
+ * We filter out all those themes because they will also be visible on
+ * second list specific to WordPress.com. This may be too simple aproach
+ * so it may revisit it again later.
+ *
+ * TODO Veriy that this aproach is sufficien.
+ *
+ * @param  {Array}  themes Array of theme objects
+ * @return {Array}         Filtered themes
+ */
+export function filterWpcomThemesFromJetpack( themes ) {
+	return filter( themes, theme => ! endsWith( theme.id, '-wpcom' ) );
+}
```

Comment:
I like the approach, but I also like modularity 😄 Can we turn this into an `isThemeOnWpcom` or `isWpcomTheme` predicate (i.e. without the filter) that we then use below in `filterThemesForJetpack`? BTW we already have an `isWporgTheme` predicate, but that is a selector. Not sure if it's better to name consistently -- i.e. `isWpcomTheme` -- or not when one is a util (no state needed) and one is a selector...